domain/src/test/java/org/oppia/android/domain/classify/rules/fractioninput/FractionInputHasIntegerPartEqualToRuleClassifierProviderTest.kt

```diff
@@ -0,0 +1,405 @@
+
+  @Test
+  fun testAnswer5Over2_input0_HasIntegerPartEqual() {
+    val inputs = mapOf("x" to WHOLE_NUMBER_0)
+
+    val matches = inputHasIntegerPartEqualToRuleClassifier.matches(
+      answer = FRACTION_5_OVER_2,
+      inputs = inputs
+    )
+
+    assertThat(matches).isTrue()
+  }
+
+  @Test
+  fun testAnswer5Over2_input2_HasNotIntegerPartEqual() {
+    val inputs = mapOf("x" to WHOLE_NUMBER_2)
+
+    val matches = inputHasIntegerPartEqualToRuleClassifier.matches(
+      answer = FRACTION_5_OVER_2,
+      inputs = inputs
+    )
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testAnswer3Over2_input2_HasNotIntegerPartEqual() {
+    val inputs = mapOf("x" to WHOLE_NUMBER_2)
+
+    val matches = inputHasIntegerPartEqualToRuleClassifier.matches(
+      answer = FRACTION_3_OVER_2,
+      inputs = inputs
+    )
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testAnswer3Over2_input3_HasNotIntegerPartEqual() {
+    val inputs = mapOf("x" to WHOLE_NUMBER_3)
+
+    val matches = inputHasIntegerPartEqualToRuleClassifier.matches(
+      answer = FRACTION_3_OVER_2,
+      inputs = inputs
+    )
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testAnswer1Over2_input1_HasNotIntegerPartEqual() {
+    val inputs = mapOf("x" to WHOLE_NUMBER_1)
+
+    val matches = inputHasIntegerPartEqualToRuleClassifier.matches(
+      answer = FRACTION_1_OVER_2,
+      inputs = inputs
+    )
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testAnswer123_1Over2_input123_HasIntegerPartEqual() {
+    val inputs = mapOf("x" to WHOLE_NUMBER_123)
+
+    val matches = inputHasIntegerPartEqualToRuleClassifier.matches(
+      answer = MIXED_NUMBER_123_1_OVER_2,
+      inputs = inputs
+    )
+
+    assertThat(matches).isTrue()
+  }
+
+  @Test
+  fun testAnswer0_2Over3_input0_HasIntegerPartEqual() {
+    val inputs = mapOf("x" to WHOLE_NUMBER_0)
+
+    val matches = inputHasIntegerPartEqualToRuleClassifier.matches(
+      answer = MIXED_NUMBER_0_2_OVER_3,
+      inputs = inputs
+    )
+
+    assertThat(matches).isTrue()
+  }
+
+  @Test
+  fun testAnswer1_2Over3_input2_HasNotIntegerPartEqual() {
+    val inputs = mapOf("x" to WHOLE_NUMBER_2)
+
+    val matches = inputHasIntegerPartEqualToRuleClassifier.matches(
+      answer = MIXED_NUMBER_1_2_OVER_3,
+      inputs = inputs
+    )
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testAnswer1_2Over3_input3_HasNotIntegerPartEqual() {
+    val inputs = mapOf("x" to WHOLE_NUMBER_3)
+
+    val matches = inputHasIntegerPartEqualToRuleClassifier.matches(
+      answer = MIXED_NUMBER_1_2_OVER_3,
+      inputs = inputs
+    )
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testAnswerNegative123_1Over2_input123_HasIntegerPartEqual() {
+    val inputs = mapOf("x" to WHOLE_NUMBER_123)
+
+    val matches = inputHasIntegerPartEqualToRuleClassifier.matches(
+      answer = NEGATIVE_MIXED_NUMBER_123_1_OVER_2,
+      inputs = inputs
+    )
+
+    assertThat(matches).isTrue()
+  }
```

Comment:
@rt4914 the sign is negligible here so that -123 and 123 returns true?

Reply:
Sounds good @MohamedMedhat1998